### PR TITLE
Refactor Balances with the help of AccountBalance

### PR DESCRIPTION
**Pull request by mjuchli-da**

### CODE REVIEW

**Review by tamaskalcza-da — APPROVED** [review]

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java [hunk 1]
```diff
@@ -5,26 +5,41 @@
...
-  public AccountInfo(Balance balance) {
-    providerParty = balance.provider;
-    iban = balance.iban;
-    assetCode = balance.currency;
-    Objects.requireNonNull(balance.owner);
-    ownerParty = balance.owner.party.orElse(null);
-    ownerName = balance.owner.name;
-  }
+    /**
+     * P   O  = A
```
Comment (darko-da) [inline gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java:33]: As soon as you are the provider it should be a liability .. it's maybe a bit of splitting hairs in this case, but I would say this should be a liability.

Comment (mjuchli-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java:33]: In this case I was thinking about the "treasury" account where a bank holds an account provided by itself...

Comment (darko-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java:33]: yes - I think we want to call this a liability

Comment (mjuchli-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java:33]: ok

**Review by tamaskalcza-da — APPROVED** [review]

**Review by darko-da — APPROVED** [review]

**Pull request merged**